app/views/milestone.py

```diff
@@ -0,0 +1,29 @@
+from flask import render_template, url_for, redirect, Blueprint, request, flash
+from flask_login import current_user, login_required
+from app.forms import MilestoneFrom
+from app.logger import log
+from app.models import User, ProjectMilestone
+from app.controllers import role_required
+
+milestone_blueprint = Blueprint("milestone", __name__)
+
+
+@milestone_blueprint.route("/milestone_add", methods=["GET", "POST"])
+@login_required
+@role_required(roles=[User.Role.project_manager])
+def milestone_add():
+    log(log.INFO, "User [%s] on milestone_add", current_user.id)
+    form = MilestoneFrom(request.form)
+    if form.validate_on_submit():
```

Comment:
are the data from the form valid? you will not make a mistake when writing data to the model or the data will be written down what you need?